### PR TITLE
Added DataDog RUM initializers to CST

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- Added DataDog RUM initializers to CST
- DD RUM is not initialized if the `cst_use_dd_rum` feature flag is false (it's false by default locally)
- DD RUM is not initialized if the user is not logged in
- files updated:
  - `src/applications/claims-status/containers/ClaimsStatusApp.jsx`
  - `src/applications/claims-status/utils/useDatadogRum.js`

## Related issue(s)

- [[CST] [DataDog RUM] [Claim List] Set up basic DataDog RUM Settings in CST](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80333)

### CODE REVIEW

File: src/applications/claims-status/utils/useDatadogRum.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  // default settings
+  site: 'ddog-gov.com',
+  // see src/site/constants/vsp-environments.js for defaults
+  env: environment.vspEnvironment(), // 'production'
+  sessionSampleRate: 5,
+  forwardErrorsToLogs: true,
+  forwardConsoleLogs: ['error'],
+  forwardReports: [],
+  telemetrySampleRate: 20, // default 20
```
Comment (jerekshoe) [inline src/applications/claims-status/utils/useDatadogRum.js:65]: I think we want both this and `sessionSampleRate` to be 100

Comment (samcoforma) [reply src/applications/claims-status/utils/useDatadogRum.js:65]: @jerekshoe No, we've been approved for different settings on CST. See Steve Alber's comment here: https://dsva.slack.com/archives/C04KHCT3ZMY/p1712607684997799?thread_ts=1712607291.510969&cid=C04KHCT3ZMY

Comment (jerekshoe) [reply src/applications/claims-status/utils/useDatadogRum.js:65]: Yeah it looks like he only specified that for session replay stuff, not for logging. I'm pretty sure we need 100% sample rates for error logging if we are going to replace Sentry with this

Comment (samcoforma) [reply src/applications/claims-status/utils/useDatadogRum.js:65]: I see, that makes sense. Waiting on confirmation from Steve and then will update. Thanks @jerekshoe

**Review by powellkerry — APPROVED** [review]

**Pull request merged**